internal/parse/parse.go

```diff
@@ -507,19 +513,30 @@ func getMounts(ctx *types.SystemContext, store storage.Store, mounts []string, c
 	//                  to allow a more robust parsing of the mount format and to give
 	//                  precise errors regarding supported format versus supported options.
 	for _, mount := range mounts {
-		arr := strings.SplitN(mount, ",", 2)
+		arr := strings.Split(mount, ",")
 		if len(arr) < 2 {
 			return nil, mountedImages, lockedTargets, fmt.Errorf("%q: %w", mount, errInvalidSyntax)
 		}
-		kv := strings.Split(arr[0], "=")
-		// TODO: type is not explicitly required in Docker.
-		// If not specified, it defaults to "volume".
-		if len(kv) != 2 || kv[0] != "type" {
-			return nil, mountedImages, lockedTargets, fmt.Errorf("%q: %w", mount, errInvalidSyntax)
+		typeToken := arr[0]
+		foundType := false
+		for _, field := range arr {
+			if strings.HasPrefix(field, "type=") {
+				foundType = true
+				typeToken = field
+			}
 		}
-
-		tokens := strings.Split(arr[1], ",")
-		switch kv[1] {
+		kv := strings.Split(typeToken, "=")
+		// If `type` is not set default to "bind"
+		mountType := TypeBind
```

Comment:
Maybe move this line to before the loop and 522 and just pull the value out of it in the loop if the argument has a "type=" prefix?  It'd probably be shorter, and `foundToken` isn't being used for much else.